pynitrokey/cli/nk3/secrets.py

```diff
@@ -0,0 +1,335 @@
+from typing import List, Optional
+
+import click
+import fido2
+
+from pynitrokey.cli.nk3 import Context, nk3
+from pynitrokey.helpers import AskUser, local_print
+from pynitrokey.nk3.secrets_app import STRING_TO_KIND, Algorithm, SecretsApp
+
+
+@nk3.group()
+@click.pass_context
+def secrets(ctx: click.Context) -> None:
+    """Nitrokey Secrets App. Manage OTP secrets on the device.
+    Use NITROPY_SECRETS_PASSWORD to pass password for the scripted execution."""
+    pass
+
+
+@secrets.command()
+@click.pass_obj
+@click.argument(
+    "name",
+    type=click.STRING,
+)
+@click.argument(
+    "secret",
+    type=click.STRING,
+    # help="The shared secret string (by default in base32)",
+)
+@click.option(
+    "--digits_str",
+    "digits_str",
+    type=click.Choice(["6", "8"]),
+    help="Digits count",
+    default="6",
+)
+@click.option(
+    "--kind",
+    "kind",
+    type=click.Choice(choices=STRING_TO_KIND.keys(), case_sensitive=False),  # type: ignore[arg-type]
+    help="OTP mechanism to use. Case insensitive.",
+    default="TOTP",
+)
+@click.option(
+    "--hash",
+    "hash",
+    type=click.Choice(["SHA1", "SHA256"]),
+    help="Hash algorithm to use",
+    default="SHA1",
+)
+@click.option(
+    "--counter_start",
+    "counter_start",
+    type=click.INT,
+    help="Starting value for the counter (HOTP only)",
+    default=0,
+)
+@click.option(
+    "--touch_button",
+    "touch_button",
+    type=click.BOOL,
+    help="This credential requires button press before use",
+    is_flag=True,
+)
+def register(
+    ctx: Context,
+    name: str,
+    secret: str,
+    digits_str: str,
+    kind: str,
+    hash: str,
+    counter_start: int,
+    touch_button: bool,
+) -> None:
+    """Register OTP credential.
+
+    Write SECRET under the NAME.
+    SECRET should be encoded in base32 format.
+    """
+    digits = int(digits_str)
+    secret_bytes = b32decode(secret)
+    otp_kind = STRING_TO_KIND[kind.upper()]
+    hash_algorithm = Algorithm.Sha1 if hash == "SHA1" else Algorithm.Sha256
+    with ctx.connect_device() as device:
+        app = SecretsApp(device)
+        ask_to_touch_if_needed()
+        authenticate_if_needed(app)
+        app.register(
+            name.encode(),
+            secret_bytes,
+            digits,
+            kind=otp_kind,
+            algo=hash_algorithm,
+            initial_counter_value=counter_start,
+            touch_button_required=touch_button,
+        )
+
+
+def check_experimental_flag(experimental: bool) -> None:
+    """Helper function to show common warning for the experimental features"""
+    if not experimental:
+        local_print(" ")
+        local_print(
+            "This feature is experimental, which means it was not tested thoroughly.\n"
+            "Note: data stored with it can be lost in the next firmware update.\n"
+            "Please pass --experimental switch to force running it anyway."
+        )
+        local_print(" ")
+        raise click.Abort()
```

Comment:
Could be dropped?

Reply:
I left it for the future experimental updates. Shall we remove it completely then?

Reply:
Leaving as is for now. Can be removed in the future.